linode/manager
# upcoming: [M3-7580] - Add search field to Create Linode UIs

## Description 📝
Adds a search field to the *Clone Linode* and *Create Linode from Backup* flows to allow users to filter Linodes based on their labels.

These changes are gated behind the `linodeCloneUIChanges` feature flag.

## Changes  🔄

- Add search field to Create Linode Clone and Backup flows
- When a user selects "Clone" on an existing Linode, navigate to Clone Linode UI, filter on the Linode's label and pre-select that Linode
- Refactored `DebouncedSearchTextField` to allow field state to be externally managed
    - This was necessary to allow automatically filtering from the query param

## Preview 📷

| Before  | After   |
| ------- | ------- |
|  <video src="https://github.com/linode/manager/assets/122488130/30d92b17-af29-434f-9dcc-81f2c724c57e" /> | <video src="https://github.com/linode/manager/assets/122488130/1ea08984-8095-4a67-bb49-def9aa8ef502" /> |

## How to test 🧪

1. Navigate to the Create Linode page
2. Select the "Clone Linode"/"Backups" tab
3. Ensure you can filter Linodes based on their label
4. Click on a Linode's action menu and select "Clone Linode"
5. Verify the Linode is selected in the list and the search field is pre-filled with the Linode's label
6. Verify you can clear and edit the contents of the search field after it is pre-filled
7. Verify [existing usages](https://github.com/linode/manager/pull/10088#discussion_r1460014481) of `DebouncedSearchTextField` continue to work as expected

## As an Author I have considered 🤔

*Check all that apply*
